Watchman.Discord/Areas/Help/Services/HelpService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Models;
+using Newtonsoft.Json;
+using Watchman.Cqrs;
+using Watchman.Discord.Areas.Help.Factories;
+using Watchman.DomainModel.Help.Models;
+using Watchman.DomainModel.Help.Queries;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.Discord.Areas.Help.Services
+{
+    public class HelpService : IService
+    {
+        private readonly IQueryBus _queryBus;
+        private readonly HelpInformationFactory _helpInformationFactory;
+        private readonly ISession _session;
+
+        public HelpService(ISessionFactory sessionFactory, IQueryBus queryBus, HelpInformationFactory helpInformationFactory)
+        {
+            _queryBus = queryBus;
+            _helpInformationFactory = helpInformationFactory;
+            _session = sessionFactory.Create();
+        }
+
+        //todo: tutaj myślę czy nie wydzielić generateHelp i generateJsonHelp do osobnej klasy
+        public string GenerateHelp(Contexts contexts)
+        {
+            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
+
+            var messageBuilder = new StringBuilder();
+            foreach (var helpInfo in result.HelpInformations)
+            {
+                var line = new StringBuilder("-" + helpInfo.Names.Aggregate((x, y) => $"{x} / -{y}"));
+                line.Append(" => ");
+                line.Append(helpInfo.Descriptions.First(x => x.Name == helpInfo.DefaultDescriptionName).Details);
+                messageBuilder.AppendLine(line.ToString());
+            }
+
+            return messageBuilder.ToString();
+        }
+
+        public IEnumerable<string> GenerateJsonHelp(Contexts contexts)
+        {
+            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
+            var serialized = JsonConvert.SerializeObject(result.HelpInformations, Formatting.Indented);
+            var smallerMessages = SplitJsonToSmallerMessages(serialized);
+
+            foreach (var sm in smallerMessages)
+            {
+                var messageBuilder = new StringBuilder();
+                messageBuilder.AppendLine("```json");
+                messageBuilder.AppendLine(sm);
+                messageBuilder.AppendLine("```");
+                yield return messageBuilder.ToString();
+            }
+        }
+
+        private IEnumerable<string> SplitJsonToSmallerMessages(string message)
+        {
+            const int MAX_MESSAGE_LENGTH = 1998; // for safety reason I made it smaller
+
+            while (message.Length > MAX_MESSAGE_LENGTH)
+            {
+                var cutIndex = message.Substring(0, MAX_MESSAGE_LENGTH).LastIndexOf("},", StringComparison.Ordinal) + 2;
```

Comment:
A co jeśli wykryje koniec obiektu na liscie argumentów 
a nie koniec informacji o komendzie?
(obiekt w środku obiektu)

Może ok opcja to pokombinowanie z regexem

Reply:
Faktycznie, spróbuję z regexem